cocos/core/assets/mesh.ts

```diff
@@ -466,18 +467,18 @@ const v3_2 = new Vec3();
 export class Mesh extends Asset {
 
     get _nativeAsset (): ArrayBuffer {
-        return this._data!.buffer;
+        return this._data.buffer;
     }
 
     set _nativeAsset (value: ArrayBuffer) {
-        if (this._data && this._data.byteLength === value.byteLength) {
-            this._data.set(new Uint8Array(value));
+        const dataU8 = new Uint8Array(value);
+        if (this._data.byteLength === value.byteLength) {
+            this._data.set(dataU8);
             if (legacyCC.loader._cache[this.nativeUrl]) {
                 legacyCC.loader._cache[this.nativeUrl].content = this._data.buffer;
```

Comment:
@holycanvas  I'm uncertain for here. Did you mean that: if the new/old `Uint8Array` have same byte length, in stead of replace the old with new, reassign the content old one and **redirect the cache** `legacyCC.loader._cache[this.nativeUrl].content` into old one's? But the type of  old `_data` is `Uint8Array`, which may be only a part of `ArrayBuffer` not whole array.

Reply:
This is for the deferred loading case, An empty data with the correct length will be initialized for use during the delayed load, and then set to this empty data after the real data is loaded. And then the data in the loader need to be updated, otherwise it will take up two copies of memory (one in this._data and one in the the loader)

Reply:
`Uint8Array` does not copy memories. It is just a view of the `ArrayBuffer`.

Reply:
As a result: https://github.com/cocos-creator/engine/pull/7131/commits/323ea58925987c6746903aa1712f0f83c364ccfc

Reply:
This conversation is invalidated.